moviepy/video/io/ImageSequenceClip.py

```diff
@@ -62,7 +62,7 @@ def __init__(
         if isinstance(sequence, list):
             if isinstance(sequence[0], str):
                 if load_images:
-                    sequence = [imread(file) for file in sequence]
+                    sequence = [imread(file, index=0)[0] for file in sequence]
```

Comment:
```suggestion
                    sequence = [img for img in iio.imiter(file)]
```

We now have [`imiter`](https://imageio.readthedocs.io/en/stable/_autosummary/imageio.v3.imiter.html#imageio.v3.imiter) which can iterate over containers and give you images in the order in which they appear in the file. 

This not only reads nicer, but is also more performant because (1) you only need to open the file once, and (2) you tell ImageIO "I want to read more than one image, just not all at once". This way we can optimize for this scenario and do things like buffering future frames/images, carry decoder state over to the next frame, select more optimized threadding models, etc. In a nutshell: we can optimize for reading and serving images sequentially.


That said, it seems odd to read all images and then store them in a list, because it can be memory hungry. Where is `sequence` consumed? Perhaps it is better to move the generator there and load images one at a time.